src/main/java/ch/njol/skript/classes/data/BukkitClasses.java

```diff
@@ -1304,15 +1304,43 @@ public boolean mustSyncDeserialization() {
 						return false;
 					}
 				}));
-
+		
 		Classes.registerClass(new ClassInfo<>(Metadatable.class, "metadataholder")
 				.user("metadata holders?")
 				.name("Metadata Holder")
 				.description("Something that can hold metadata (e.g. an entity or block)")
 				.examples("set metadata value \"super cool\" of player to true")
-				.since("2.2-dev36")
-		);
-
+				.since("2.2-dev36"));
+		
+		Classes.registerClass(new ClassInfo<>(CachedServerIcon.class, "cachedservericon")
+				.user("((loaded|cached) ?)?server ?icons?")
+				.name("Loaded Server Icon")
+				.description("A server icon that loaded using the <a href='effects.html#EffLoadServerIcon'>load server icon</a> effect.")
+				.examples("")
+				.since("INSERT VERSION")
+				.parser(new Parser<CachedServerIcon>() {
+					@Override
+					@Nullable
+					public CachedServerIcon parse(final String s, final ParseContext context) {
+						return null;
+					}
+					
+					@Override
+					public String toString(final CachedServerIcon o, int flags) {
+						return "server icon";
+					}
+					
+					@SuppressWarnings("null")
+					@Override
+					public String toVariableNameString(final CachedServerIcon o) {
+						return "server icon";
+					}
+					
+					@Override
+					public String getVariableNamePattern() {
+						return "\\S+";
```

Comment:
this won't match

Reply:
so what does that mean? If i do ``set {x::%last loaded server icon%} to 1`` it is ``{x::server icon}``

Reply:
it's used for variable conflict warnings, it is meant to return a pattern that will match all the potential string forms from `Parser#toVariableNameString`. In this case, i think you could just return `server icon` as that pattern will match `server icon`, the only string form a CachedServerIcon.